examples/imagenet/alex.py

```diff
@@ -37,3 +40,30 @@ def __call__(self, x, t):
         loss = F.softmax_cross_entropy(h, t)
         chainer.report({'loss': loss, 'accuracy': F.accuracy(h, t)}, self)
         return loss
+
+
+class Alex_fp16(Alex):
+
+    """Single-GPU AlexNet without partition toward the channel axis."""
+
+    insize = 227
+
+    def __init__(self):
+        self.dtype = np.float16
+        W = initializers.HeNormal(1 / np.sqrt(2), self.dtype)
+        bias = initializers.Zero(self.dtype)
+        chainer.Chain.__init__(
+            self,
+            conv1=L.Convolution2D(3,  96, 11, stride=4, initialW=W, bias=bias),
+            conv2=L.Convolution2D(96, 256,  5, pad=2, initialW=W, bias=bias),
+            conv3=L.Convolution2D(256, 384,  3, pad=1, initialW=W, bias=bias),
+            conv4=L.Convolution2D(384, 384,  3, pad=1, initialW=W, bias=bias),
+            conv5=L.Convolution2D(384, 256,  3, pad=1, initialW=W, bias=bias),
+            fc6=L.Linear(9216, 4096, initialW=W, bias=bias),
+            fc7=L.Linear(4096, 4096, initialW=W, bias=bias),
+            fc8=L.Linear(4096, 1000, initialW=W, bias=bias),
+        )
+        self.train = True
+
+    def __call__(self, x, t):
+        return Alex.__call__(self, F.cast(x, self.dtype), t)
```

Comment:
I think that current code is easy to understand. Users may want simple example.
How do you think?